spring-xd-dirt/src/main/java/org/springframework/xd/dirt/plugins/stream/ModuleTypeConversionPlugin.java

```diff
@@ -142,14 +166,14 @@ private MimeType resolveContentType(String type, Module module) throws ClassNotF
 		return MimeType.valueOf(type);
 	}
 
-	private Class<?> resolveJavaType(String type, Module module) throws ClassNotFoundException, LinkageError {
+	private static Class<?> resolveJavaType(String type, Module module) throws ClassNotFoundException, LinkageError {
 		SimpleModule sm = (SimpleModule) module;
 		return ClassUtils.forName(type, sm.getApplicationContext().getClassLoader());
 	}
 
 	@Override
 	public boolean supports(Module module) {
-		return true;
+		return module.shouldBind();
```

Comment:
I made a comment on the pre-PR commit that was similar, but my point was that "shouldBind()" is not quite capturing what this boolean indicates now that it's expanded to this as well. The fundamental challenge here is that the MessageConverters cannot be created on the XD container node, because they are not serializable (the base class has no public default constructor and that is in core Spring, not XD so we cannot workaround it). That means the converters have to be created within the actual Spark executor process where they will be used. That is why the plugin code is being reused from the lifecycle methods (on start) in those instances that have themselves been serialized and shipped to the cluster nodes. @ilayaperumalg can surely provide more detail, but that's my quick overview of the situation.